examples/map/map_segment.py

```diff
@@ -0,0 +1,118 @@
+"""
+=======================================================
+Segmenting a Map based on transformation of coordinates
+=======================================================
+
+A demostration creating a segment of a particular map from
+transformed coordinates.
+"""
+import numpy as np
+import matplotlib.pyplot as plt
+from reproject import reproject_interp
+
+import astropy.units as u
+from astropy.wcs import WCS
+from astropy.coordinates import SkyCoord
+
+import sunpy.map
+from sunpy.coordinates import Helioprojective, NorthOffsetFrame, RotatedSunFrame, transform_with_sun_center
+from sunpy.data.sample import AIA_171_ROLL_IMAGE
+
+###############################################################################
+# We start with the sample data.
+smap = sunpy.map.Map(AIA_171_ROLL_IMAGE)
+
+###############################################################################
+# A utility function gives us access to the helioprojective coordinate of each
+# pixel. From this we then transform the coordinates to the required frame.
+all_hpc = sunpy.map.all_coordinates_from_map(smap)
+all_hgs = all_hpc.transform_to("heliographic_stonyhurst")
+
+###############################################################################
+# Let's then segment the data based on coordinates and convert them to pixel
+# index. We convert the segmented pixel values to 0.
+segment = np.logical_and(all_hgs.lon < 10 * u.deg, all_hgs.lon > 0 * u.deg)
+
+###############################################################################
+# Let's plot the segment separately
+new_data = np.asarray(segment, dtype='float')
+new_data = np.where(new_data == False, np.nan, new_data)
+
+new_frame_map = sunpy.map.Map(new_data * smap.data, smap.meta)
+fig = plt.figure()
+fig.add_subplot(projection=smap)
+new_frame_map.plot()
+plt.show()
+
+###############################################################################
+# This example can also be extended while using NorthOffsetFrame
+# Let us offset the north pole and create the frame
+north = SkyCoord(20 * u.deg, 20 * u.deg, frame="heliographic_stonyhurst")
+f = NorthOffsetFrame(north=north)
+
+###############################################################################
+# We then transform coordinates to the Offsetted frame and segment the data
+# based on conditions
+all_hpc = sunpy.map.all_coordinates_from_map(smap)
+ofsetted_coords = all_hpc.transform_to(f)
+segment = np.logical_and(
+    ofsetted_coords.lon < 30 * u.deg,
+    ofsetted_coords.lon > -20 * u.deg)
+
+###############################################################################
+# Let's plot the ofsetted segment separately
+new_data = np.asarray(segment, dtype='float')
+new_data[new_data == False] = np.nan
+
+ofsetted_map = sunpy.map.Map(new_data * smap.data, smap.meta)
+fig = plt.figure()
+fig.add_subplot(projection=smap)
+ofsetted_map.plot()
+plt.show()
+
+###############################################################################
```

Comment:
Now I see this, I wonder if it's worth having this in here. I wonder if a section on doing more mathematical operations with the mask, like sum / mean / etc instead.

Reply:
Should I do that instead? Like add another part to this example performing mathematical operations.
Do we keep the `NorthOffsetFrame` example though?

Reply:
I think maybe just drop the `RotatedSunFrame` stuff and add some examples of using the mask in operations to the end of both the other sections.